examples/react-router/app/components/recipe_adder.jsx

```diff
@@ -0,0 +1,66 @@
+var React = require("react"),
+    Router = require("react-router"),
+    RouteHandler = Router.RouteHandler,
+    Link = Router.Link,
+    Navigation = Router.Navigation,
+    State = Router.State,
+    Fluxxor = require("../../../../");
+
+var RecipeForm = require("../forms/recipe_form.jsx");
+
+var RecipeAdder = React.createClass({
+  mixins: [
+    Fluxxor.FluxMixin(React),
+    State,
+    Navigation
+  ],
+
+  render: function() {
+    return this.renderWithLayout(
+      <div>
+        <form onSubmit={this.onSubmit}>
+          <RecipeForm ref="form" />
+          <input type="submit" value="Save" />
+        </form>
+      </div>
+    );
+  },
+
+  renderWithLayout: function(content) {
+    return (
+      <div>
+        {content}
+        <hr />
+        <Link to="home">Home</Link>
+        {" | "}<Link to="add-recipe">Add New Recipe</Link>
+      </div>
+    );
+  },
+
+  onSubmit: function(e) {
+    e.preventDefault();
+
+    var newRecipe = this.refs.form.getValue();
+    if (newRecipe) {
+      this.getFlux().actions.recipes.add(
+        newRecipe.name,
+        newRecipe.description,
+        newRecipe.ingredients,
+        newRecipe.directions,
+        function(recipe) {
```

Comment:
I have taken different approach (mixed feelings about it) in side project. My approach is following: I'm getting _pending_ item from the Store and watch if it has id in `getStateFromFlux` so component is handling redirect, but I've stumbled into problems with that:
- transition in `getStateFromFlux` has to be wrapped in `setTimeout` because next transition could trigger actions
```
    if(store.pending.id && !store.pending.loading) {
      setTimeout(() => {
        this.transitionTo('Workspaces.Show', { workspace_id: store.pending.id });
      }, 0)
    }
```
- Works only new items, for editing I've to watch for errors and maintain state in component. 

I like though because it keeps data flow one way and I can observe in different component if something is currently created or has been added. I'm was also thinking of creating new mixin that would allow bindings into custom events emitted from store because `change` is sometimes too general.